storage/aptosdb/src/lib.rs

```diff
@@ -1293,7 +1297,12 @@ impl DbWriter for AptosDB {
                         .map_or(-1, |c| c as i64),
                 );
 
-                self.wake_pruner(last_version);
+                if *self.last_version_sent_to_pruner.lock()
+                    < last_version - self.versions_per_pruning_batch
+                {
+                    self.wake_pruner(last_version);
+                    *self.last_version_sent_to_pruner.lock() = last_version;
+                }
```

Comment:
shouldn't this be managed by the pruner itself?

Reply:
Yes, I thought about this but we also want to avoid serializing command and waking up the pruner for every version, right?

Reply:
Actually, I take it back, we serialize the command inside the pruner, so we can move it in the pruner without much performance overhead. Let me do that.